src/PrivateKey.cpp

```diff
@@ -162,17 +164,20 @@ Data PrivateKey::sign(const Data& digest, TWCurve curve) const {
         result.resize(64);
         const auto publicKey = getPublicKey(TWPublicKeyTypeED25519);
         ed25519_sign(digest.data(), digest.size(), bytes.data(), publicKey.bytes.data(), result.data());
+        success = true;
     } break;
     case TWCurveED25519Blake2bNano: {
         result.resize(64);
         const auto publicKey = getPublicKey(TWPublicKeyTypeED25519Blake2b);
         ed25519_sign_blake2b(digest.data(), digest.size(), bytes.data(),
                              publicKey.bytes.data(), result.data());
+        success = true;
     } break;
     case TWCurveED25519Extended: {
         result.resize(64);
         const auto publicKey = getPublicKey(TWPublicKeyTypeED25519Extended);
         ed25519_sign_ext(digest.data(), digest.size(), bytes.data(), extensionBytes.data(), publicKey.bytes.data(), result.data());
+        success = true;
```

Comment:
does `ed25519_sign_ext` function success by itself? maybe we should check that

Reply:
That was my first thought as well, but some trezor-crypto sign methods have bool return value, some have none.  I guess that's why the original optimistic success = true initialization.  It is safer to start out with false.

void ed25519_sign_ext(const unsigned char *m, size_t mlen, const ed25519_secret_key sk, const ed25519_secret_key skext, const ed25519_public_key pk, ed25519_signature RS);